pkg/k8s/labels.go

```diff
@@ -82,6 +82,69 @@ const (
 	// in service identity.
 	IdentityModeAnnotation = Prefix + "/identity-mode"
 
+	/*
+	 * Proxy config annotations
+	 */
+
+	// ProxyImageAnnotation can be used to override the proxyImage config.
+	ProxyImageAnnotation = "proxy.linkerd.io/proxy-image"
```

Comment:
We should talk about the format of this label in more detail. I had originally anticipated all linkerd-related annotations would be prefixed with "linkerd.io" and this format breaks that assumption.